### ISSUE TITLE
Long tail class inmixing

### ISSUE THREAD

**Issue by guuske31** [GaoLii/DSP#9]
Hello, first of all thanks for the contribution and sharing the code. I have been using this implementation for a while and considering the class "rand_mixer" for mixing the long tail classes with source/target inputs, I notice that the long tail classes tend to not show up in the mixed result. To ilustrate the issue, I add some screenshots of plots
for each variable.

Considering the code, I believe that the long tail class mixing in line 325 in train.py:
```python
   data, target = transformsgpu.oneMix(MixMask, data=mixdata, target=mixtarget)
```

I introduced plotting the data and targets before and after this function to see the long tail mixing in action with the following code:
```python
      plot_input(MixMask, 'mask_lt', 'Mask over the long tail class image', save=True)
      plot_input(img, 'image_lt', "Image where the long tail class is in", save=True)
      plot_input(lbl, 'label_lt', 'Label of the longtail class image', save=True)

      data, target = transformsgpu.oneMix(MixMask, data=mixdata, target=mixtarget)

      plot_input(mixdata[1], 'image_lt', 'Image where the long tail class comes in', save=True)
      plot_input(data[0], 'image_lt', 'Mixed image', save=True)

      plot_input(mixtarget[1], 'label_lt', 'Label where the long tail class comes in', save=True)
      plot_input(target[0], 'label_lt', 'Mixed label', save=True)
```

The plot input functions that is used above, is shown here:
```python
   def plot_input(input, input_type, name=None, save=False):
    with torch.no_grad():

        if input_type == 'image_lt':  # Image while in long tail function
            input = input.cpu().detach().numpy()

            input = np.transpose(input, (1, 2, 0))

            image = input + IMG_MEAN
            image = image[:, :, ::-1]
 
            plt.imshow(image.astype(int))


        elif input_type == 'mask_lt': # mask while in long tail function

            input = input.cpu().detach().numpy()
            plt.imshow(input, cmap='binary')

        elif input_type == 'label_lt': # label while in long tail function

            input = input.cpu().detach().numpy()
            color_mask = cm_eval(input)
            # print(color_mask)
            plt.imshow(color_mask)

        else:
            print("Incorrect input for creating an image")

        if not (name is None):
            title_obj = plt.title(name)  # get the title property handler
            plt.getp(title_obj)

        if save:
            start_writeable = datetime.datetime.now().strftime('%m-%d_%H-%M')
            plt.savefig(os.path.join("./saved/images", start_writeable+input_type+name), bbox_inches='tight')
        plt.show()
```

So when I plot this. The following figures are displayed:
- The Mask over the long tail image:
![12-14_10-56mask_ltMask over the long tail class image](https://user-images.githubusercontent.com/50746212/145982560-7d0f91af-d4be-4f3b-8ada-223a7ffa1376.png)

- The Image with the long tail class object in it:
![12-14_10-56image_ltImage where the long tail class is in](https://user-images.githubusercontent.com/50746212/145982891-9da6de0a-2cdf-4048-b149-38bd68b1ae48.png)

- The Label with the long tail class object in it:
![12-14_10-56label_ltLabel of the longtail class image](https://user-images.githubusercontent.com/50746212/145982960-e8726bb9-3f33-48d3-9595-9e74b37d95e1.png)

- The Image where the long tail class object comes into:
![12-14_10-57image_ltImage where the long tail class comes in](https://user-images.githubusercontent.com/50746212/145983666-615fcf43-2b9e-4f35-846c-e3ecfa8bda34.png)

- The Label where the long tail class object comes into:
![12-14_10-57label_ltLabel where the long tail class comes in](https://user-images.githubusercontent.com/50746212/145983724-b4d04876-07fc-4f21-8fa5-02ffa20cc529.png)

- The Image with the long tail class object mixed in:
![12-14_10-57image_ltMixed image](https://user-images.githubusercontent.com/50746212/145983620-52f59839-37e2-4d9c-8ea9-24de281396e2.png)

- The Label with the long tail class object mixed in:
![12-14_10-57label_ltMixed label](https://user-images.githubusercontent.com/50746212/145983585-5c03679c-2874-4f15-aff4-88bdf0feeee6.png)

As you can see, the long tail class is not mixed into the image. This happens in every occasion, but in many situations the long tail class is really small, so not visible in the image. Above a clear example is shown. 

Countering the argument above, I have had two times a situation, where the image with the long tail class objects was overwritten in the image and label partly. However, this was done in a small strip in the vertical direction. An additional requirement for this to occur, was that the long tail class object was at the (x,0) position. I have established to save one of those examples. These are shown below:

- The Mask over the long tail image:
![12-14_10-52mask_ltMask over the long tail class image](https://user-images.githubusercontent.com/50746212/145986185-d3532775-b640-4e64-90ee-eed7f4d85d20.png)

- The Image with the long tail class object in it:
![12-14_10-52image_ltImage where the long tail class is in](https://user-images.githubusercontent.com/50746212/145986246-f5fef6ec-d256-445f-bb08-cccae6ffb3eb.png)

- The Label with the long tail class object in it:
![12-14_10-52label_ltLabel of the longtail class image](https://user-images.githubusercontent.com/50746212/145986264-94526ade-5d99-4408-b795-f47c1cea1edd.png)

- The Image where the long tail class object comes into:
![12-14_10-52image_ltImage where the long tail class comes in](https://user-images.githubusercontent.com/50746212/145986294-3c8f7ee4-6ded-4787-98a0-62d262fb9131.png)

- The Label where the long tail class object comes into:
![12-14_10-52label_ltLabel where the long tail class comes in](https://user-images.githubusercontent.com/50746212/145986316-461843f7-b454-4387-a868-ef5d2b4c86d7.png)

- The Image with the long tail class object mixed in:
![12-14_10-52image_ltMixed image](https://user-images.githubusercontent.com/50746212/145986343-a6e720f1-05ac-4490-b152-06e6c73041f6.png)

- The Label with the long tail class object mixed in:
![12-14_10-52label_ltMixed label](https://user-images.githubusercontent.com/50746212/145986356-c3b0d31b-4ad3-42e3-8872-6cf3b091381d.png)

Could someone explain this behavior to me?
And if possible suggest a solution to this problem.
